x/btcstaking/keeper/msg_server.go

```diff
@@ -220,18 +220,22 @@ func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCre
 		return nil, types.ErrReusedStakingTx.Wrapf("duplicated tx hash: %s", stakingTxHash.String())
 	}
 
-	// 4. Check finality providers to which message delegate
-	// Ensure all finality providers are known to Babylon, are not slashed
-	for _, fpBTCPK := range parsedMsg.FinalityProviderKeys.PublicKeysBbnFormat {
-		// get this finality provider
-		fp, err := ms.GetFinalityProvider(ctx, fpBTCPK)
-		if err != nil {
-			return nil, err
-		}
-		// ensure the finality provider is not slashed
-		if fp.IsSlashed() {
-			return nil, types.ErrFpAlreadySlashed.Wrapf("finality key: %s", fpBTCPK.MarshalHex())
-		}
+	// verify proof of possession
+	if err := req.Pop.Verify(parsedMsg.StakerAddress, req.BtcPk, ms.btcNet); err != nil {
```

Comment:
duplicated pop verification ?

Pop is already verified above in point two:
```

	// 2. Basic stateless checks
	// - verify proof of possession
	if err := parsedMsg.ParsedPop.Verify(parsedMsg.StakerAddress, parsedMsg.StakerPK.BIP340PubKey, ms.btcNet); err != nil {
		return nil, types.ErrInvalidProofOfPossession.Wrap(err.Error())
	}
```

Reply:
Yeah this is duplicated -- with the new `ParseCreateDelegationMessage` we don't need to do PoP verification anymore.